src/script.js

```diff
@@ -22,19 +22,42 @@
       return ((v2Rect.width * v2Rect.height) - (v1Rect.width * v1Rect.height));
     });
 
-  if (videos.length === 0)
+  if (videos.length === 0) {
     return;
+  }
+
+  return videos[0];
+}
 
-  const video = videos[0];
+async function requestPictureInPicture(video) {
+  await video.requestPictureInPicture();
+  video.setAttribute('__pip__', true);
+  video.addEventListener('leavepictureinpicture', event => {
+    video.removeAttribute('__pip__');
+  }, { once: true });
+}
+
+async function maybeUpdatePictureInPictureVideo() {
+  if (!document.querySelector('[__pip__]')) {
+    return;
+  }
+  const video = findLargestPlayingVideo();
+  if (video && !video.hasAttribute('__pip__')) {
+    await requestPictureInPicture(video);
+  }
+}
 
+(async () => {
+  const video = findLargestPlayingVideo();
+  if (!video) {
+    return;
+  }
   if (video.hasAttribute('__pip__')) {
-    await document.exitPictureInPicture();
-  } else {
-    await video.requestPictureInPicture();
-    video.setAttribute('__pip__', true);
-    video.addEventListener('leavepictureinpicture', event => {
-      video.removeAttribute('__pip__');
-    }, { once: true });
-    chrome.runtime.sendMessage({ message: 'enter' });
+    document.exitPictureInPicture();
+    return;
   }
+  await requestPictureInPicture(video);
+  chrome.runtime.sendMessage({ message: 'enter' });
+
+  new ResizeObserver(maybeUpdatePictureInPictureVideo).observe(video);
```

Comment:
Shouldn't we update the video we observer when `maybeUpdatePictureInPictureVideo` is called? See suggestion above.

Reply:
We should probably save the ResizeObserver object too, right?